runtime/gc_glue_java/MarkingDelegate.cpp

```diff
@@ -322,6 +322,7 @@ MM_MarkingDelegate::completeMarking(MM_EnvironmentBase *env)
 									clazz = javaVM->internalVMFunctions->hashClassTableNextDo(&walkState);
 								}
 
+#if JAVA_SPEC_VERSION > 8
```

Comment:
Similar to above but you would have to remove the `Assert` call.

Reply:
I opted to use runtime check `JAVA_SPEC_VERSION > 8` instead. Technically `classLoader->moduleHashTable` should not present in `Java 8` code, and there is chance to remove it via compile time check in VM side.

@charliegracie @dmitripivkine the PR is ready for another review.

Reply:
I think for now it would be better to use the check I suggested as it is more future proof. if you remove the field then it has to switch to a compile time check.

Reply:
> I think for now it would be better to use the check I suggested as it is more future proof.

Sure, on the other hand, do you like `if ((JAVA_SPEC_VERSION > 8) && (NULL != classLoader->moduleHashTable)) {` instead? In this case, even the field `moduleHashTable` is removed  for `Java 8`, `(JAVA_SPEC_VERSION > 8)` still guards the following code block.

Reply:
Updated the PR to check `(NULL != classLoader->moduleHashTable)` and removed `Assert_MM_true(NULL != classLoader->moduleHashTable)`.
Please have another look.

Just a separated question, is it ok to add compile time check `#if JAVA_SPEC_VERSION > 8` in GC code when the field `moduleHashTable` or others are removed from `Java 8`?